analyzer/metadata_registry/metadata_registry.go

```diff
@@ -0,0 +1,90 @@
+package metadata_registry
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	registry "github.com/oasisprotocol/metadata-registry-tools"
+	staking "github.com/oasisprotocol/oasis-core/go/staking/api"
+
+	"github.com/oasisprotocol/nexus/analyzer"
+	"github.com/oasisprotocol/nexus/analyzer/item"
+	"github.com/oasisprotocol/nexus/analyzer/queries"
+	"github.com/oasisprotocol/nexus/common"
+	"github.com/oasisprotocol/nexus/log"
+	"github.com/oasisprotocol/nexus/storage"
+)
+
+const MetadataRegistryAnalyzerName = "metadata_registry"
+
+type processor struct {
+	target storage.TargetStorage
+	logger *log.Logger
+}
+
+var _ item.ItemProcessor[common.ChainName] = (*processor)(nil)
+
+func NewAnalyzer(
+	batchSize uint64,
+	stopOnEmptyBatch bool,
+	fixedInterval time.Duration,
+	target storage.TargetStorage,
+	logger *log.Logger,
+) (analyzer.Analyzer, error) {
+	logger.Info("Starting metadata_registry analyzer")
+	if fixedInterval == 0 {
+		fixedInterval = 2 * time.Minute
+	}
+	if fixedInterval < time.Minute {
+		return nil, fmt.Errorf("invalid interval %s, metadata registry interval must be at least 1 minute", fixedInterval)
+	}
+	logger = logger.With("analyzer", MetadataRegistryAnalyzerName)
+	p := &processor{
+		target: target,
+		logger: logger,
+	}
+
+	return item.NewAnalyzer[common.ChainName](
+		MetadataRegistryAnalyzerName,
+		batchSize,
+		stopOnEmptyBatch,
+		fixedInterval,
+		p,
+		target,
+		logger,
+	)
+}
+
+func (p *processor) GetItems(ctx context.Context, limit uint64) ([]common.ChainName, error) {
+	return []common.ChainName{common.ChainNameMainnet}, nil
```

Comment:
note: for this and the node_stats analyzer, this fixed queue means that these analyzers are not yet compatible with the e2e_regression tests. This will likely need to be slightly reworked with`stopOnEmptyBatch` but I'd prefer to leave it for a future PR as this one is already somewhat long, and the regression tests do not currently test those analyzers so there is no loss of functionality.

Reply:
does this fully coincide with the use of `fixedInterval`?

Reply:
> does this fully coincide with the use of fixedInterval?

currently yes